frontend/src/v5/ui/components/shared/modals/templates/deleteModal/deleteModal.component.tsx

```diff
@@ -14,48 +14,113 @@
  *  You should have received a copy of the GNU Affero General Public License
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import { Button, DialogContent, DialogContentText, DialogTitle } from '@mui/material';
+import { Button, DialogContentText, DialogTitle } from '@mui/material';
 import DeleteIcon from '@assets/icons/delete.svg';
 import { FormattedMessage } from 'react-intl';
-import { ModalContainer, Actions } from '@/v5/ui/components/shared/modals/modals.styles';
+import {
+	DialogContainer,
+	Actions,
+	RetypeCheck,
+	ConfirmationPhrase,
+	RetypeCheckField,
+	Message,
+	TruncatableTitle,
+	Instruction,
+	ErrorMessage,
+} from '@/v5/ui/components/shared/modals/modals.styles';
 import { CircledIcon } from '@controls/circledIcon';
+import { formatMessage } from '@/v5/services/intl';
+import { useForm } from 'react-hook-form';
+import { useState } from 'react';
+import { isNetworkError } from '@/v5/validation/errors.helpers';
 
 interface IDeleteModal {
 	onClickClose?: () => void,
-	onClickConfirm: () => void,
-	title: string,
+	onClickConfirm: (onSuccess, onError) => any,
```

Comment:
If we change onClickConfirm to be a promise it could be easier to follow the flow
```suggestion
	onClickConfirm: () => Promise<void>,
```